### DISCUSSION TITLE
How to override default theme?

### DISCUSSION THREAD

**Discussion by fredrik-sogaard** [chakra-ui/chakra-ui#6933] [Q&A]
This seems like noob question, but how do I _override_ the default theme, and not _extend_ it? I want to remove all for example all the font weights other than those I specify:

```
const Theme = extendTheme({
  fontWeights: {
      thin: 200,
      normal: 400,
      bold: 700,
      xBold: 800,
    }
})
```
If I do this, I still end up with all of Chakra UI's font weights. How do I get rid of the default ones?

In Tailwind the default is to override the keys, and we have the option to `extend`:

```
theme: {
    fontWeight: {
      normal: 400,
      bold: 700,
    },
    extend: {
      spacing: {
        '8xl': '96rem',
        '9xl': '128rem',
      },
    }
  },
```

Is this possible with Chakra?

**Comment by anddoutoi** [comment]
There are basically 2 things you can do.

### Not using `extendTheme`…

and provide the `<ChakraProvider>` with your "raw" theme object:
```js
const theme = {
  fontWeights: {
    thin: 200,
    normal: 400,
    bold: 700,
    xBold: 800,
    },
};

<ChakraProvider theme={theme}>
```
I guess however you want to keep some styles from Chakra UIs default theme?

### Using `extendTheme`…

`extendTheme()` uses the last argument you provide the function with as "baseTheme". If you don't provide a "baseTheme" yourself, like only providing the function with 1 argument, it WILL use Chakra UIs default theme as "baseTheme" to extend from. See current flow: https://github.com/chakra-ui/chakra-ui/blob/7a1716aab5405c71180e9a0a3f1123782c18c3df/packages/utilities/theme-utils/src/extend-theme.ts#L86-L97

Using this knowledge you can create your own baseTheme from Chakra UIs default theme with an empty fontWeights key and extend from that.

```js
import defaultTheme from "@chakra-ui/theme";

const yourOwnBaseThemeToExtendFrom = {
  ...defaultTheme,
  fontSizes: {}
};

const theme = extendTheme(
  {
    fontWeights: {
      thin: 200,
      normal: 400,
      bold: 700,
      xBold: 800,
    },
  },
  yourOwnBaseThemeToExtendFrom
);
```
The above is not tested. I only [followed the code](https://github.com/chakra-ui/chakra-ui/blob/7a1716aab5405c71180e9a0a3f1123782c18c3df/packages/components/theme/src/index.ts) and made hypothetical conclusions.

**Reply by fredrik-sogaard** [reply]
Thanks for the reply!  I works with your suggested first method: provide Chakra with a raw theme object.

However the second method didn't work, I end up with them same result as with just extending the theme as normal.
